### PR TITLE
CAS-451 Ephemeral message's footnote

### PR DESCRIPTION
**Pull request by samiuelson**
https://stream-io.atlassian.net/browse/CAS-451

### Description

Display ephemeral message "footnote"

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
~- [ ] Changelog updated with client-facing changes~
~- [ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TimeDecorator.kt [hunk 1]
```diff
@@ -5,54 +5,66 @@ import androidx.core.view.isVisible
...
             data.positions.contains(MessageListItem.Position.BOTTOM).not() || createdAt == null -> textView.isVisible = false
             else -> {
-                textView.isVisible = true
-                textView.text = dateFormatter.formatTime(createdAt)
+                textView.apply {
+                    isVisible = true
+                    val createdAtTime = dateFormatter.formatTime(createdAt)
+                    if (data.message.isEphemeral()) {
+                        val footerTemplate = context.getString(R.string.stream_ui_ephemeral_msg_footer_template)
+                        text = String.format(footerTemplate, createdAtTime)
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TimeDecorator.kt:61]: Wouldn't it make things easier if this footnote message was a separate view? For example, we could add spacing between it and the timestamp, and that sort of thing.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TimeDecorator.kt [hunk 2]
```diff
@@ -5,54 +5,66 @@ import androidx.core.view.isVisible
...
+                        setCompoundDrawablesWithIntrinsicBounds(R.drawable.stream_ui_ic_icon_eye_off, 0, 0, 0)
+                        compoundDrawablePadding = resources.getDimensionPixelSize(R.dimen.stream_ui_spacing_small)
+                    } else {
+                        text = createdAtTime
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TimeDecorator.kt:65]: Don't we need to clear the drawable on this branch?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/DeliveryStatusDecorator.kt [hunk 3]
```diff
@@ -88,5 +89,9 @@ internal class DeliveryStatusDecorator : BaseDecorator() {
                 hideIndicator()
             }
         }.exhaustive
+
+        if (data.message.isEphemeral()){
```
Comment (samiuelson) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/DeliveryStatusDecorator.kt:93]: Hiding indicator for ephemeral messages

Comment (zsmb13) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/DeliveryStatusDecorator.kt]: We should do these amongst the checks in the beginning of the method

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_file_attachments.xml [hunk 4]
```diff
@@ -44,26 +44,44 @@
...
-        android:layout_marginEnd="4dp"
-        app:layout_constraintBottom_toBottomOf="@+id/tvTime"
-        app:layout_constraintEnd_toStartOf="@+id/tvTime"
-        app:layout_constraintTop_toTopOf="@+id/tvTime"
-        tools:src="@drawable/stream_ui_ic_check_all"
-        />
-
-    <TextView
-        android:id="@+id/tvTime"
+    <androidx.appcompat.widget.LinearLayoutCompat
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_file_attachments.xml:47]: Seems like a lot of doubling. What if we extract it to a standalone layout file and use it through the `include` mechanism? Or even we could create a custom view and put this to the required places.

**Review by ogkuzmin — CHANGES_REQUESTED** [review]
Idea is clear and good. But I'd like to remove doublings

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_file_attachments.xml (hunk 4 above)
Comment (samiuelson) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_file_attachments.xml]: IMO, include will work fine for now.

**Review by ogkuzmin — APPROVED** [review]
LGTM

**Pull request merged**